### ISSUE TITLE
Math error in macOS ARM version of DOSBox?

### ISSUE THREAD

**Issue by digistorm79** [joncampbell123/dosbox-x#3387]
### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

### Have you checked that no other similar bug report(s) already exists?

- [X] I have searched and didn't find any similar issues.

### What operating system(s) this bug have occurred on?

macOS Monterey 12.3.1 running on M1 Pro

### What version(s) of DOSBox-X have this bug?

0.83.24 macOS ARM build

### Describe the bug

When running the PC Player Benchmark from PhilsComputerLabs DOSBench pack, it does not apply the camera position. The whole scene runs viewed straight from above. It looks like it miscalculates or ignores a certain calculation during the 3D transform. This seems to correlate with other reported bugs of Quake not launching and some FPU bug when running Windows 98.
![pcpbench_000](https://user-images.githubusercontent.com/45852917/161501029-30f6e71c-5d8b-42e1-a8d9-61d13ee1acf0.png)

### Expected behavior

On the Intel build, running with Rosetta 2, the benchmark runs as expected. Or any other machine running DOSBox, or any real PC.
![pcpbench_001](https://user-images.githubusercontent.com/45852917/161501043-59c8bb66-afe4-4b42-bdb2-99cc9158325e.png)

### Steps to reproduce the behaviour

1. Download the DOSBench suite from PhilsComputerLab (https://www.philscomputerlab.com/uploads/3/7/2/3/37231621/dosbench_v_1.4_jan_2017.zip)
2. Extract it and execute it by running DOSBENCH.BAT
3. Type 6 to run the benchmark
4. It looks weird and wrong on the ARM build, and correct on any other build

### Emulator log

```text
It outputs nothing that relates to this benchmark, only messages about changing the output resolution.
```

### Additional context

The bug appears with any type of core and any type of emulated CPU.

**Comment by joncampbell123** [comment]
This may be related to Touhou Project 1 where the orb is stuck in the air, rolling left to right, and never falls downward.

**Comment by digistorm79** [comment]
**Update:**
In the current version (0.84.0) the benchmark - just like Quake - works perfectly with the "normal" cpu core. With the dynamic core, it crashes with a black screen, just like a lot of other applications since version 0.84.0.

**Comment by CyberXZT** [comment]
I believe I may be seeing this same bug on the Raspberry Pi 4. Tested with the latest commit and 0.84.2. I also tested with a couple older versions. I could have sworn I had Quake running on the Raspberry Pi 4 with the dynamic core but none of the versions I compiled worked. Some failed in ways different to the below.

Here are my tests:

Normal core:
- **Chris's 3D Benchmark** - Works
- **PC Player Benchmark** - Works
- **Quake** - Works
- **Landmark** - Works

Dynamic core:
- **Chris's 3D Benchmark** - Fails to start and returns a score of "-214748364.-8" or "0.0".
- **PC Player Benchmark** - Freezes on black screen. Switching between normal and dynamic cores during run causes camera problems or a crash
- **Quake** - Fails to start
- **Landmark** - Displays invalid numbers for CPU Clock and Video speed. Gets stuck in an infinite loop attempting to draw CPU and FPU information

Dynamic core, FPU disabled:
- **Chris's 3D Benchmark** - Same as dynamic
- **PC Player Benchmark** - Freezes on black screen
- **Quake** - Same as dynamic
- **Landmark** - Works

Dynamic core, SDL2:
- **Chris's 3D Benchmark** - segfault
- **PC Player Benchmark** - Same as dynamic
- **Quake** - segfault
- **Landmark** - segfault

I've also attached logs of Quake starting with the dynamic core and the normal core:

[quake-dynamic.txt](https://github.com/joncampbell123/dosbox-x/files/9530801/quake-dynamic.txt)
[quake-normal.txt](https://github.com/joncampbell123/dosbox-x/files/9530803/quake-normal.txt)

Amusing Note: It seems Quake will try to open a file named "emu387.dxe". I tried a few versions that I found on the web but have not been able to make it do anything.

**Comment by digistorm79** [comment]
It is fixed in version 2023.03.31
